packages/@o3r/core/schematics/ng-add/project-setup/index.ts

```diff
@@ -0,0 +1,93 @@
+import { chain, noop, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { NodeDependencyType } from '@schematics/angular/utility/dependencies';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import { PackageJson } from 'type-fest';
+import {
+  createAzurePipeline,
+  generateRenovateConfig,
+  o3rBasicUpdates,
+  updateAdditionalModules,
+  updateCmsAdapter,
+  updateCustomizationEnvironment,
+  updateFixtureConfig,
+  updateLinter,
+  updateOtterEnvironmentAdapter,
+  updatePlaywright,
+  updateStore
+} from '../../rule-factories/index';
+import { NgAddSchematicsSchema } from '../schema';
+import { updateBuildersNames } from '../updates-for-v8/cms-adapters/update-builders-names';
+import { updateOtterGeneratorsNames } from '../updates-for-v8/generators/update-generators-names';
+import { packagesToRemove } from '../updates-for-v8/replaced-packages';
+
+const simplifiedSemVerRegexp = new RegExp(/^[~^]?(0|(?:[1-9]+[0-9]*))\.(?:0|[1-9]+[0-9]*)\.(?:0|[1-9]+[0-9]*)(?:-[a-zA-Z]+\.(?:0|[1-9]+[0-9]*))?$/);
```

Comment:
why not using `semver` library to extract the major version instead?

Reply:
I did not want to add a new dependency just for a regexp